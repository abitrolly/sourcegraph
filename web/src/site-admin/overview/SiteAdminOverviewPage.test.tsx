import * as H from 'history'
import React from 'react'
import { of } from 'rxjs'
import { SiteAdminOverviewPage } from './SiteAdminOverviewPage'
import { eventLogger } from '../../tracking/eventLogger'
import sinon from 'sinon'
import { ISiteUsagePeriod } from '../../../../shared/src/graphql/schema'
import { PageTitle } from '../../components/PageTitle'
import { mount } from 'enzyme'

describe('SiteAdminOverviewPage', () => {
    afterEach(() => {
        PageTitle.titleSet = false
    })

    const baseProps = {
        history: H.createMemoryHistory(),
        isLightTheme: true,
        overviewComponents: [],
    }

    let stub: sinon.SinonStub<[string, (boolean | undefined)?], void>

    beforeAll(() => {
        stub = sinon.stub(eventLogger, 'logViewEvent')
    })

    afterAll(() => {
        if (stub) {
            stub.restore()
        }
    })

    test('activation in progress', () => {
        expect(
            mount(
                <SiteAdminOverviewPage
                    {...baseProps}
                    activation={{
                        steps: [
                            {
                                id: 'ConnectedCodeHost' as const,
                                title: 'Add repositories',
                                detail: 'Configure Sourcegraph to talk to your code host',
                            },
                        ],
                        completed: {
                            ConnectedCodeHost: false,
                        },
                        update: sinon.stub(),
                        refetch: sinon.stub(),
                    }}
                    _fetchOverview={() =>
                        of({
                            repositories: 0,
                            users: 1,
                            orgs: 1,
                            surveyResponses: {
                                totalCount: 0,
                                averageScore: 0,
                            },
                        })
                    }
                    _fetchWeeklyActiveUsers={() =>
                        of({
                            __typename: 'SiteUsageStatistics',
                            daus: [],
                            waus: [],
                            maus: [],
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })

    test('< 2 users', () => {
        expect(
            mount(
                <SiteAdminOverviewPage
                    {...baseProps}
                    _fetchOverview={() =>
                        of({
                            repositories: 100,
                            users: 1,
                            orgs: 1,
                            surveyResponses: {
                                totalCount: 1,
                                averageScore: 10,
                            },
                        })
                    }
                    _fetchWeeklyActiveUsers={() =>
                        of({
                            __typename: 'SiteUsageStatistics',
                            daus: [],
                            waus: [],
                            maus: [],
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })
    test('>= 2 users', () => {
        const usageStat: ISiteUsagePeriod = {
            __typename: 'SiteUsagePeriod',
            userCount: 10,
            registeredUserCount: 8,
            anonymousUserCount: 2,
            integrationUserCount: 0,
            startTime: new Date().toISOString(),
            stages: undefined as any,
        }
        expect(
            mount(
                <SiteAdminOverviewPage
                    {...baseProps}
                    _fetchOverview={() =>
                        of({
                            repositories: 100,
                            users: 10,
                            orgs: 5,
                            surveyResponses: {
                                totalCount: 100,
                                averageScore: 10,
                            },
                        })
                    }
                    _fetchWeeklyActiveUsers={() =>
                        of({
                            __typename: 'SiteUsageStatistics',
                            daus: [],
                            waus: [usageStat, usageStat],
                            maus: [],
                        })
                    }
                />
            )
        ).toMatchSnapshot()
    })
})
