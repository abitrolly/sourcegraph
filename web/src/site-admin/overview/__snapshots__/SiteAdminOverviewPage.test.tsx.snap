// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteAdminOverviewPage < 2 users 1`] = `
<SiteAdminOverviewPage
  _fetchOverview={[Function]}
  _fetchWeeklyActiveUsers={[Function]}
  history="[History]"
  isLightTheme={true}
  overviewComponents={Array []}
>
  <div
    className="site-admin-overview-page"
  >
    <PageTitle
      title="Overview - Admin"
    />
    <div
      className="pt-3 mb-4"
    />
    <div
      className="list-group"
    >
      <AnchorLink
        className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
        to="/site-admin/repositories"
      >
        <a
          className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
          href="/site-admin/repositories"
        >
          100
           
          repositories
        </a>
      </AnchorLink>
    </div>
  </div>
</SiteAdminOverviewPage>
`;

exports[`SiteAdminOverviewPage >= 2 users 1`] = `
<div
  className="site-admin-overview-page"
>
  <div
    className="pt-3 mb-4"
  />
  <div
    className="list-group"
  >
    <a
      className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
      href="/site-admin/repositories"
    >
      100
       
      repositories
    </a>
    <a
      className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
      href="/site-admin/users"
    >
      10
       
      users
    </a>
    <a
      className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
      href="/site-admin/organizations"
    >
      5
       
      organizations
    </a>
    <a
      className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
      href="/site-admin/surveys"
    >
      100
       
      user survey responses
    </a>
    <div
      className="list-group-item"
    >
      <div
        className="d-flex justify-content-between align-items-center position-relative"
      >
        <span
          className="h5 mb-0 font-weight-normal p-2"
        >
          10
           
          active users
           last week
        </span>
        <button
          aria-label="Collapse section"
          className="d-flex btn btn-icon collapsible__expand-btn stretched-link"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-label="Close section"
            className="mdi-icon icon-inline"
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
            />
          </svg>
        </button>
      </div>
      <div
        className="site-admin-usage-statistics-page"
      >
        <div
          className="site-admin-overview-page__detail-header"
        >
          <h2>
            Weekly unique users
          </h2>
          <h3>
            <a
              className="btn btn-secondary"
              href="/site-admin/usage-statistics"
            >
              View all usage statistics
               
              <svg
                className="mdi-icon icon-inline"
                fill="currentColor"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M14,3V5H17.59L7.76,14.83L9.17,16.24L19,6.41V10H21V3M19,19H5V5H12V3H5C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V12H19V19Z"
                />
              </svg>
            </a>
          </h3>
        </div>
        <svg
          viewBox="0 0 500 200"
        />
        <small
          className="site-admin-usage-statistics-page__tz-note"
        >
          <i>
            GMT/UTC time
          </i>
        </small>
      </div>
    </div>
  </div>
</div>
`;

exports[`SiteAdminOverviewPage activation in progress 1`] = `
<SiteAdminOverviewPage
  _fetchOverview={[Function]}
  _fetchWeeklyActiveUsers={[Function]}
  activation={
    Object {
      "completed": Object {
        "ConnectedCodeHost": false,
      },
      "refetch": [Function],
      "steps": Array [
        Object {
          "detail": "Configure Sourcegraph to talk to your code host",
          "id": "ConnectedCodeHost",
          "title": "Add repositories",
        },
      ],
      "update": [Function],
    }
  }
  history="[History]"
  isLightTheme={true}
  overviewComponents={Array []}
>
  <div
    className="site-admin-overview-page"
  >
    <PageTitle
      title="Overview - Admin"
    />
    <div
      className="pt-3 mb-4"
    >
      <Collapsible
        buttonClassName="mb-0 py-3 px-3"
        className="p-0 list-group-item font-weight-normal e2e-site-admin-overview-menu"
        defaultExpanded={true}
        detail="Complete the steps below to finish onboarding to Sourcegraph"
        detailClassName="h5 mb-0 font-weight-normal"
        title={
          <React.Fragment>
            Welcome to Sourcegraph
          </React.Fragment>
        }
        titleAtStart={true}
        titleClassName="h5 mb-0 font-weight-bold"
      >
        <div
          className="p-0 list-group-item font-weight-normal e2e-site-admin-overview-menu"
        >
          <div
            className="d-flex justify-content-between align-items-center position-relative mb-0 py-3 px-3"
          >
            <div
              className="d-flex flex-column"
            >
              <span
                className="h5 mb-0 font-weight-bold"
              >
                Welcome to Sourcegraph
              </span>
              <div
                className="h5 mb-0 font-weight-normal"
              >
                Complete the steps below to finish onboarding to Sourcegraph
              </div>
            </div>
            <button
              aria-label="Collapse section"
              className="d-flex btn btn-icon collapsible__expand-btn stretched-link"
              onClick={[Function]}
              type="button"
            >
              <Memo(ChevronDownIcon)
                aria-label="Close section"
                className="icon-inline"
              >
                <svg
                  aria-label="Close section"
                  className="mdi-icon icon-inline"
                  fill="currentColor"
                  height={24}
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <path
                    d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                  />
                </svg>
              </Memo(ChevronDownIcon)>
            </button>
          </div>
          <ActivationChecklist
            buttonClassName="h5 mb-0 font-weight-normal"
            completed={
              Object {
                "ConnectedCodeHost": false,
              }
            }
            history="[History]"
            steps={
              Array [
                Object {
                  "detail": "Configure Sourcegraph to talk to your code host",
                  "id": "ConnectedCodeHost",
                  "title": "Add repositories",
                },
              ]
            }
          >
            <div
              className="activation-checklist list-group list-group-flush "
            >
              <Accordion
                collapsible={true}
              >
                <DescendantProvider
                  context={
                    Object {
                      "$$typeof": Symbol(react.context),
                      "Consumer": Object {
                        "$$typeof": Symbol(react.context),
                        "_calculateChangedBits": null,
                        "_context": Object {
                          "$$typeof": Symbol(react.context),
                          "Consumer": Object {
                            "$$typeof": Symbol(react.context),
                            "_calculateChangedBits": null,
                            "_context": Object {
                              "$$typeof": Symbol(react.context),
                              "Consumer": Object {
                                "$$typeof": Symbol(react.context),
                                "_calculateChangedBits": null,
                                "_context": [Circular],
                              },
                              "Provider": Object {
                                "$$typeof": Symbol(react.provider),
                                "_context": [Circular],
                              },
                              "_calculateChangedBits": null,
                              "_currentRenderer": Object {},
                              "_currentRenderer2": null,
                              "_currentValue": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_currentValue2": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_threadCount": 0,
                              "displayName": "AccordionDescendantContext",
                            },
                          },
                          "Provider": Object {
                            "$$typeof": Symbol(react.provider),
                            "_context": Object {
                              "$$typeof": Symbol(react.context),
                              "Consumer": Object {
                                "$$typeof": Symbol(react.context),
                                "_calculateChangedBits": null,
                                "_context": [Circular],
                              },
                              "Provider": Object {
                                "$$typeof": Symbol(react.provider),
                                "_context": [Circular],
                              },
                              "_calculateChangedBits": null,
                              "_currentRenderer": Object {},
                              "_currentRenderer2": null,
                              "_currentValue": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_currentValue2": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_threadCount": 0,
                              "displayName": "AccordionDescendantContext",
                            },
                          },
                          "_calculateChangedBits": null,
                          "_currentRenderer": Object {},
                          "_currentRenderer2": null,
                          "_currentValue": Object {
                            "descendants": Array [],
                            "registerDescendant": [Function],
                            "unregisterDescendant": [Function],
                          },
                          "_currentValue2": Object {
                            "descendants": Array [],
                            "registerDescendant": [Function],
                            "unregisterDescendant": [Function],
                          },
                          "_threadCount": 0,
                          "displayName": "AccordionDescendantContext",
                        },
                      },
                      "Provider": Object {
                        "$$typeof": Symbol(react.provider),
                        "_context": Object {
                          "$$typeof": Symbol(react.context),
                          "Consumer": Object {
                            "$$typeof": Symbol(react.context),
                            "_calculateChangedBits": null,
                            "_context": Object {
                              "$$typeof": Symbol(react.context),
                              "Consumer": Object {
                                "$$typeof": Symbol(react.context),
                                "_calculateChangedBits": null,
                                "_context": [Circular],
                              },
                              "Provider": Object {
                                "$$typeof": Symbol(react.provider),
                                "_context": [Circular],
                              },
                              "_calculateChangedBits": null,
                              "_currentRenderer": Object {},
                              "_currentRenderer2": null,
                              "_currentValue": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_currentValue2": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_threadCount": 0,
                              "displayName": "AccordionDescendantContext",
                            },
                          },
                          "Provider": Object {
                            "$$typeof": Symbol(react.provider),
                            "_context": Object {
                              "$$typeof": Symbol(react.context),
                              "Consumer": Object {
                                "$$typeof": Symbol(react.context),
                                "_calculateChangedBits": null,
                                "_context": [Circular],
                              },
                              "Provider": Object {
                                "$$typeof": Symbol(react.provider),
                                "_context": [Circular],
                              },
                              "_calculateChangedBits": null,
                              "_currentRenderer": Object {},
                              "_currentRenderer2": null,
                              "_currentValue": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_currentValue2": Object {
                                "descendants": Array [],
                                "registerDescendant": [Function],
                                "unregisterDescendant": [Function],
                              },
                              "_threadCount": 0,
                              "displayName": "AccordionDescendantContext",
                            },
                          },
                          "_calculateChangedBits": null,
                          "_currentRenderer": Object {},
                          "_currentRenderer2": null,
                          "_currentValue": Object {
                            "descendants": Array [],
                            "registerDescendant": [Function],
                            "unregisterDescendant": [Function],
                          },
                          "_currentValue2": Object {
                            "descendants": Array [],
                            "registerDescendant": [Function],
                            "unregisterDescendant": [Function],
                          },
                          "_threadCount": 0,
                          "displayName": "AccordionDescendantContext",
                        },
                      },
                      "_calculateChangedBits": null,
                      "_currentRenderer": Object {},
                      "_currentRenderer2": null,
                      "_currentValue": Object {
                        "descendants": Array [],
                        "registerDescendant": [Function],
                        "unregisterDescendant": [Function],
                      },
                      "_currentValue2": Object {
                        "descendants": Array [],
                        "registerDescendant": [Function],
                        "unregisterDescendant": [Function],
                      },
                      "_threadCount": 0,
                      "displayName": "AccordionDescendantContext",
                    }
                  }
                  items={
                    Array [
                      Object {
                        "disabled": false,
                        "element": <button
                          aria-controls="panel--1--0"
                          aria-expanded="false"
                          class="activation-checklist__button list-group-item list-group-item-action"
                          data-reach-accordion-button=""
                          id="button--1--0"
                        >
                          <div
                            class="activation-checklist-item d-flex justify-content-between h5 mb-0 font-weight-normal"
                          >
                            <div
                              class="d-flex align-items-center"
                            >
                              <span
                                class="activation-checklist-item__icon-container icon-inline icon-down"
                              >
                                <svg
                                  class="mdi-icon activation-checklist-item__icon"
                                  fill="currentColor"
                                  height="24"
                                  viewBox="0 0 24 24"
                                  width="24"
                                >
                                  <path
                                    d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                                  />
                                </svg>
                              </span>
                              <span
                                class="activation-checklist-item__icon-container icon-inline icon-right"
                              >
                                <svg
                                  class="mdi-icon activation-checklist-item__icon"
                                  fill="currentColor"
                                  height="24"
                                  viewBox="0 0 24 24"
                                  width="24"
                                >
                                  <path
                                    d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                                  />
                                </svg>
                              </span>
                              <a
                                class="activation-checklist__title"
                                href=""
                              >
                                Add repositories
                              </a>
                            </div>
                            <div>
                              <svg
                                class="mdi-icon icon-inline text-muted"
                                fill="currentColor"
                                height="24"
                                viewBox="0 0 24 24"
                                width="24"
                              >
                                <path
                                  d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                                />
                              </svg>
                            </div>
                          </div>
                        </button>,
                        "index": 0,
                      },
                    ]
                  }
                  set={[Function]}
                >
                  <div
                    data-reach-accordion=""
                  >
                    <AccordionItem
                      className="activation-checklist__container list-group-item "
                      key="ConnectedCodeHost"
                    >
                      <div
                        className="activation-checklist__container list-group-item "
                        data-reach-accordion-item=""
                        data-state="collapsed"
                      >
                        <AccordionButton
                          className="activation-checklist__button list-group-item list-group-item-action"
                        >
                          <button
                            aria-controls="panel--1--0"
                            aria-expanded={false}
                            className="activation-checklist__button list-group-item list-group-item-action"
                            data-reach-accordion-button=""
                            id="button--1--0"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                          >
                            <ActivationChecklistItem
                              className="h5 mb-0 font-weight-normal"
                              detail="Configure Sourcegraph to talk to your code host"
                              done={false}
                              history="[History]"
                              id="ConnectedCodeHost"
                              key="ConnectedCodeHost"
                              title="Add repositories"
                            >
                              <div
                                className="activation-checklist-item d-flex justify-content-between h5 mb-0 font-weight-normal"
                              >
                                <div
                                  className="d-flex align-items-center"
                                >
                                  <span
                                    className="activation-checklist-item__icon-container icon-inline icon-down"
                                  >
                                    <Memo(ChevronDownIcon)
                                      className="activation-checklist-item__icon"
                                    >
                                      <svg
                                        className="mdi-icon activation-checklist-item__icon"
                                        fill="currentColor"
                                        height={24}
                                        viewBox="0 0 24 24"
                                        width={24}
                                      >
                                        <path
                                          d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                                        />
                                      </svg>
                                    </Memo(ChevronDownIcon)>
                                  </span>
                                  <span
                                    className="activation-checklist-item__icon-container icon-inline icon-right"
                                  >
                                    <Memo(ChevronRightIcon)
                                      className="activation-checklist-item__icon"
                                    >
                                      <svg
                                        className="mdi-icon activation-checklist-item__icon"
                                        fill="currentColor"
                                        height={24}
                                        viewBox="0 0 24 24"
                                        width={24}
                                      >
                                        <path
                                          d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                                        />
                                      </svg>
                                    </Memo(ChevronRightIcon)>
                                  </span>
                                  <a
                                    className="activation-checklist__title"
                                    href=""
                                  >
                                    Add repositories
                                  </a>
                                </div>
                                <div>
                                  <Memo(CheckboxBlankCircleOutlineIcon)
                                    className="icon-inline text-muted"
                                  >
                                    <svg
                                      className="mdi-icon icon-inline text-muted"
                                      fill="currentColor"
                                      height={24}
                                      viewBox="0 0 24 24"
                                      width={24}
                                    >
                                      <path
                                        d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                                      />
                                    </svg>
                                  </Memo(CheckboxBlankCircleOutlineIcon)>
                                </div>
                              </div>
                            </ActivationChecklistItem>
                          </button>
                        </AccordionButton>
                        <AccordionPanel
                          className="px-2"
                        >
                          <div
                            aria-labelledby="button--1--0"
                            className="px-2"
                            data-reach-accordion-panel=""
                            data-state="collapsed"
                            hidden={true}
                            id="panel--1--0"
                            role="region"
                            tabIndex={-1}
                          >
                            <div
                              className="activation-checklist__detail pb-1"
                            >
                              Configure Sourcegraph to talk to your code host
                            </div>
                          </div>
                        </AccordionPanel>
                      </div>
                    </AccordionItem>
                  </div>
                </DescendantProvider>
              </Accordion>
            </div>
          </ActivationChecklist>
        </div>
      </Collapsible>
    </div>
    <div
      className="list-group"
    >
      <AnchorLink
        className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
        to="/site-admin/repositories"
      >
        <a
          className="list-group-item list-group-item-action h5 mb-0 font-weight-normal py-2 px-3"
          href="/site-admin/repositories"
        >
          0
           
          repositories
        </a>
      </AnchorLink>
    </div>
  </div>
</SiteAdminOverviewPage>
`;
